Yoast/wordpress-seo
# Backfill options

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the removed options are no longer available for other plugins to use. These are reintroduced using a backfill, which fetches the data that is expected for the old options from the new locations.

## Relevant technical choices:

* Introduced a new class to contain all backfilling of the options and user meta.
* This class is needed on both frontend and backend, thus placed in the `inc` folder with the option framework.
* Removed the backfill hooks before the `WPSEO_Options::get_lookup` method to make sure the right settings are retrieved from the options.

## Test instructions

This PR can be tested by following these steps:

* Use `get_option()` with the specific option names and make sure the contents is what you expect it to be.
* Removed options: 'wpseo_xml', 'wpseo_rss', 'wpseo_permalinks', 'wpseo_internallinks'
* Modified options: 'wpseo', 'wpseo_titles'
* User meta field `wpseo_excludeauthorsitemap` should retrieve the value from: `wpseo_noindex_author`

The `wpseo_xml` option has been split up in two parts. Some settings have been moved to other 
options:
- 'enablexmlsitemap'       => 'enable_xml_sitemap',
- 'disable_author_sitemap' => 'noindex-author-wpseo',
- 'disable_author_noposts' => 'noindex-author-noposts-wpseo',

others have been replaced by filters:
- `entries-per-page` = `wpseo_sitemap_entries_per_page`
- `excluded-posts` = `wpseo_exclude_from_sitemap_by_post_ids`

The `wpseo_permalinks` option has a couple of settings that have been removed completely:
- 'cleanpermalinks'
- 'cleanpermalink-extravars'
- 'cleanpermalink-googlecampaign'
- 'cleanpermalink-googlesitesearch'
- 'cleanreplytocom'
- 'cleanslugs'
- 'trailingslash'

Other settings have been moved to another option:
- 'redirectattachment' => 'disable-attachment',
- 'stripcategorybase'  => 'stripcategorybase',

The `wpseo_rss` and `wpseo_internallinks` have been moved entirely to other options.

## Quality assurance


Fixes #9127
